src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +10,22 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserException("User cannot be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("User with this login already exists");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidUserException("Login must contain 6 or more symbols");
```

Comment:
Consider splitting this if statement into two separate checks for `user.getLogin() == null` and `user.getLogin().length() < 6`. This will make the code more readable and allow for more precise error messages, which is advisable as per the checklist.